core/psh/psh.c

```diff
@@ -1174,7 +1291,14 @@ int main(int argc, char **argv)
 		}
 		/* Run shell interactively */
 		else {
-			psh_run();
+			psh_run(1);
+		}
+	}
+	else if (!strcmp(base, "pshlogin")) {
+		int code = -1;
+		while ((code = psh_login()) != 0) {
+			if (code > 0)
+				psh_run(0);
```

Comment:
I'm not sure the "exitable" approach is the way to go (also: I could probably still "exit" by other means - by ^D or explicitly killing myself).

Probably the correct way would be to spawn new `psh` process when the old dies (either by kernel-space init thread or preferably by userspace init process (which could be psh)). We can also go for spawning the child psh after logging in and assume for now that the parent "pshlogin" would never die (for now, we need to ensure that at a later time by one of the previously mentioned approaches).

This "feature" explicitly disallows us to "logout" which is a crucial part of logging procedure.

Reply:
Yes. It is not the most elegant way. However, in my opinion, it is the more simple and readable if we want to stick to psh login feature to be applet-like.

As for now it may be time consuming to ensure both:
- not being able to _brick_ the system by exiting/killing psh or some psh_login process explicitly
- being able to exit psh process (as it seems reasonable to be able to kill some shell process)

only in psh and not by other means with respect to resource consumption. And it may become obsolete when new ways of ensuring that will appear. Thats why i stayed with the `exitable` solution.

Reply:
as we discussed, this is acceptable for now due to constrained environment we're running psh in